src/main/groovy/wooga/gradle/build/unity/tasks/UnityBuildPlayerTask.groovy

```diff
@@ -160,34 +171,22 @@ class UnityBuildPlayerTask extends AbstractUnityProjectTask {
     protected void exec() {
         File out = getOutputDirectory()
         out.mkdirs()
-        String customArgs = "-CustomArgs:platform=${getBuildPlatform()};"
-        customArgs += "environment=${getBuildEnvironment()};"
+        String customArgs = "-CustomArgs:appConfig=${getAppConfigFile()};"
         customArgs += "outputPath=${out.getPath()};"
         customArgs += "version=${getVersion()};"
 
+        if (getBuildPlatform()) {
+            args BatchModeFlags.BUILD_TARGET, getBuildPlatform()
```

Comment:
So given that it's a string and Unity doesn't serialize nulls in scriptable objects most likely it will be an empty string, not missing and not null. I think we need to check for `getBuildPlatform` also being not an empty string.

And above there's a test testing no build target provided I think it should also test for this case. A build target being empty string.

Reply:
see [a580a3e](https://github.com/wooga/atlas-build-unity/pull/16/commits/a580a3e4193a522c617c497598da98684efb7f7d) with test spec for this case.